### ISSUE TITLE
the ratio of classification loss and verification loss is 1/2

### ISSUE THREAD

**Issue by JoyLuo** [layumi/2016_person_re-ID#4]
dear @layumi 
  when i train the siamese net, the loss of verification shakes widly. and it is down in a very difficult way to understand. 
  (1)how about you view of the change(down) of the verification loss 
  (2)do you have tune the net in other ratio of the two loss, and could you tell me the result
  and this is the fig of classification loss and verification loss
![clsloss](https://cloud.githubusercontent.com/assets/16474619/25259606/b6b80364-2679-11e7-94a8-f6fd9ae94289.PNG)
![difloss](https://cloud.githubusercontent.com/assets/16474619/25259611/bb463b8a-2679-11e7-9a98-d9ee8ab74180.PNG)
  and thanks for you kindly

**Comment by layumi** [comment]
@JoyLuo 
Hi Joy,
Sorry for the late response. 
1. The 'shake' of the verification loss may be due to the (x-y)^2 in the Square layer.
I have tried to add L2-norm Layer but it seems to harm the training process and loss some information.   I did not try adding the BatchNorm, and I think it may work.
  
2. Actually I have tried classification:verification = 1:0.1,1:0.01 and the result is similar or no better than 1:1.  So I do not try more variants.

Thank you! 
Zhedong

**Comment by JoyLuo** [comment]
@layumi thanks for you kindly answer. 
     could you show me the figure of loss_dif your trained? it will tell me the right way about how to down the loss_dif.

thank you!
luoze

**Comment by layumi** [comment]
I just uploaded the training curve of the Market-1501. You may refer to it. 
https://github.com/layumi/2016_person_re-ID/blob/master/market1501-train-curve.pdf

The top1_compare and objective_compare are for the verification loss.

**Comment by JoyLuo** [comment]
dear @layumi 
    i see the curve from you link. thanks for you kindly help
